src/Microsoft.Extensions.Hosting.Abstractions/BackgroundService.cs

```diff
@@ -62,14 +64,34 @@ public virtual Task StartAsync(CancellationToken cancellationToken)
             finally
             {
                 // Wait until the task completes or the stop token triggers
-                await Task.WhenAny(_executingTask, Task.Delay(Timeout.Infinite, cancellationToken));
-            }
+                var task = await Task.WhenAny(_executingTask, Task.Delay(Timeout.Infinite, cancellationToken));
```

Comment:
You still haven't given a good reason for this to be in a finally block.

Reply:
It's because throwing on Cancel doesn't mean the task itself has unwound. Cancellation happens on a different thread and the executing task is running on.

Reply:
You already abandon the task if the other CT trips. How is that much different from the shutdown CT throwing? May may also end up masking an exception.

Reply:
Its a best effort. The callback might throw because it was handed off to another component that throw not because the background task itself throws. The goal was to avoid just leaving a dangling task. It's a best effort.

> You may also end up masking an exception.

Masking it how?

Reply:
By masking I mean that Cancel can throw and then _executingTask can throw and you lose the exception from Cancel.

Reply:
Yea, I get that, if we were being pedantic we could throw an aggregate exception but I dunno if it matters